plugin/ocgrpc/client.go

```diff
@@ -29,6 +29,11 @@ type ClientHandler struct {
 	// StartOptions.SpanKind will always be set to trace.SpanKindClient
 	// for spans started by this handler.
 	StartOptions trace.StartOptions
+
+	// NoPropagate may be set to true to avoid propagating tags and tracing
+	// on outbound calls. This is appropriate to set on gRPC calls to external
+	// services.
+	NoPropagate bool
```

Comment:
I think I didn't understand the goal of this feature initially. As far as I understand now, not propagating the tags is a security feature. We don't want to propagate tags to untrusted endpoints.

What about having a field here called `IsPublicEndpoint` and not propagation if it is true? So it is more symmetric to the handler field.

Reply:
I don't think "public endpoint" makes sense in the context of a client, only a server. I would be open to deprecating/renaming `IsPublicEndpoint` to `NoPropagate` -- I think that makes sense for both, what do you think?

Reply:
Why not? The outgoing URL can be called a public/insecure endpoint. It makes less sense for a server because it is not an endpoint but more of a starting point.

Reply:
That doesn't agree with my understanding of the word "endpoint", but anyway I removed this part of the PR. I think it is better handled by having custom propagation format support for gRPC (like we do for HTTP) and instead using the same technique of a `None()` propagation format that can be configured.  This will make gRPC and HTTP align better.